tt_metal/common/core_coord.h

```diff
@@ -45,6 +47,17 @@ constexpr inline bool operator==(const RelativeCoreCoord &a, const RelativeCoreC
 
 constexpr inline bool operator!=(const RelativeCoreCoord &a, const RelativeCoreCoord &b) { return !(a == b); }
 
+namespace std {
+template <>
+struct hash<RelativeCoreCoord> {
+    std::size_t operator()(RelativeCoreCoord const &o) const {
+        std::size_t seed = 0;
+        seed = std::hash<std::size_t>()(o.x) ^ std::hash<std::size_t>()(o.y) << 1;
```

Comment:
It is not a good hash for a grid coordinates.
It will return same hash for every coordinate where y == x/2. 
Larger shift might be much better (or just if x == y return x; else return x ^ y; will do the trick) because I believe x and y are not so huge numbers.
But if there is no performance concerns we can live with it.